resource/ui/ts/components/select.ts

```diff
@@ -0,0 +1,45 @@
+// source: components/select.ts
+/// <reference path="../typings/mithriljs/mithril.d.ts" />
+// Author: Matt Tracy (matt@cockroachlabs.com)
+//
+
+/**
+ * Components defines reusable components which may be used on multiple pages,
+ * or multiple times on the same page.
+ */
+module Components {
+	/**
+	 * Select is a basic html option select.
+	 */
+	export module Select {
+		import property = _mithril.MithrilProperty;
+
+		export interface Item {
+			value: string;
+			text: string;
+		}
+
+		export interface Options {
+			items: Item[];
+			selected: property<string>;
+			callback: (string) => void;
+		}
+
+		export function controller(options: Options): Options {
+
+			return options;
+		}
+
+		export function view(ctrl: Options) {
+			var onChangeFn = function(value: string) {
```

Comment:
Looking at this again, this method really should be on the controller.

The class could be as simple as:
```
class Controller {
   constructor(public options:Options){}
   onChange = (val:string) => {
     options.selected(val);
     options.callback(val);
   }
}
```